src/main/scala/mesosphere/marathon/state/Group.scala

```diff
@@ -108,10 +109,26 @@ class Group(
 
   override def hashCode(): Int = Objects.hash(id, apps, pods, groupsById, dependencies, version)
 
-  override def toString = s"Group($id, ${apps.values}, ${pods.values}, ${groupsById.values}, $dependencies, $version)"
+  private def summarize[T](iterator: Iterator[T]): String = {
+    val s = new StringBuilder
+    s ++= "Seq("
+    s ++= iterator.take(3).toSeq.mkString(", ")
+    if (iterator.hasNext)
+      s ++= s", ... ${iterator.length} more"
+    s ++= ")"
+    s.toString
+  }
+  override def toString = {
+    val summarizedApps = summarize(apps.valuesIterator.map(_.id))
+    val summarizedPods = summarize(pods.valuesIterator.map(_.id))
+    val summarizedGroups = summarize(groupsById.valuesIterator.map(_.id))
+    val summarizedDependencies = summarize(dependencies.iterator)
+
+    s"Group($id, $summarizedApps, $summarizedPods, $summarizedGroups, $summarizedDependencies, $version)"
```

Comment:
This needs to be more clear... it's current output would be something like:
`Group(/, Seq(/marathon-slack, /marathon-lb), Seq(), Seq(/development, /integration))`

It is currently not possible from the logs to know what is apps, what is pods, what is groups, etc.
lets make this easier and more parsable.

Reply:
👍

Reply:
It would be easy enough to put it, "apps = ...", Etc. agreed.